### PR TITLE
Use magic byte instead of tag to differentiate between pre-attestation and attestation

### LINKED ISSUE #111: Simplify differentiating between pre-attestation and attestation based on magic byte
Currently we use magic byte to distinguish between consensus operation and other operations. Then further deeper inspection gives us tag value based on which we determine if this is pre-endorsement or endorsement. But simply with magic byte we can distinguish between pre-attestation and attestation, we dont need tag information.

### PR DESCRIPTION
**Pull request by ajinkyaraj-23**
fixes #111

### CODE REVIEW

File: src/baking_auth.c [hunk 1]
```diff
@@ -230,33 +226,23 @@ bool parse_block(buffer_t *buf, parsed_baking_data_t *const out) {
...
-bool parse_consensus_operation(buffer_t *buf, parsed_baking_data_t *const out) {
+bool parse_consensus_operation(buffer_t *buf,
+                               parsed_baking_data_t *const out,
+                               bool is_attestation) {
     uint8_t tag;
 
     if (!buffer_read_u32(buf, &out->chain_id.v, BE) ||   // chain id
         !buffer_seek_cur(buf, 32u * sizeof(uint8_t)) ||  // ignore branch
-        !buffer_read_u8(buf, &tag) ||                    // tag
+        !buffer_read_u8(buf, &tag) ||                    // ignore tag
```
Comment (spalmer25) [inline src/baking_auth.c:236]: ignore the `tag` using `!buffer_seek_cur(buf, sizeof(uint8_t))`

**Review by spalmer25 — COMMENTED** [review]
I do not understand why we have to do this changes.
Does checking the attestation/pre-attestation tag leads to some issues ?

File: src/baking_auth.c (hunk 1 above)
Comment (ajinkyaraj-23) [reply src/baking_auth.c]: Done.

**Comment by ajinkyaraj-23** [comment]
Its a minor optimization. It came up in discussion in Proto-ecosystem team where they mentioned that tag is not necessary for signers.

**Review by spalmer25 — APPROVED** [review]

**Pull request merged**